File: ruby/birthday-cake-candles.rb
def birthdayCakeCandles(candles)
  highest_number = candles.reduce(0) do |max, number|
    if number > max
      number
    else
      max
    end
  end
  count = 0
  candles.each do |candle|
    if candle == highest_number
      count += 1
    end
  end
  return count
end

File: Python/birthday-cake-candles.py
def birthdayCakeCandles(candles):
  index = 0
  max = candles[0]
  while index < len(candles):
    if candles[index] > max:
      max = candles[index]
    index += 1
  count, index = 0, 0
  while index < len(candles):
    if candles[index] == max:
      count += 1
    index += 1
  else:
    return count